robot/board.py

```diff
@@ -41,6 +42,14 @@ def __init__(self, socket_path):
         self.data = b''
 
         self._connect(socket_path)
+        self._serial = Path(socket_path).stem
+
+    @property
+    def serial(self):
```

Comment:
Since this property does nothing but expose the `_serial` variable, I would just make the variable a public attribute and document is in the docstring for the class (as per [PEP 257](https://www.python.org/dev/peps/pep-0257/#id17)).

Reply:
The main benefit of storing it as an internal variable is it's immutable, and means there's little chance of them accidentally overriding it.

Reply:
Hmm, yeah, that seems reasonable enough. I would be tempted then to make the `serial` property just call `self.socket_path.stem` and not store it as a `_serial` attribute. If that's slow (which I don't think it is) there is always [@cached_property](https://github.com/pydanny/cached-property).

Reply:
I literally just made that change ;)